### PR TITLE
Create CF Good Subarrays.mdx

### PR DESCRIPTION
**Pull request by jessechoe10**

### CODE REVIEW

File: solutions/CF Good Subarrays.mdx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
```
Comment (maggie-j-liu) [inline solutions/CF Good Subarrays.mdx:1]: edit [this file](https://github.com/cpinitiative/usaco-guide/blob/master/content/3_Silver/Prefix_Sums.problems.json#L128) to include this internal solution by changing `kind` to `internal` and removing `site`

File: solutions/CF Good Subarrays.mdx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+[Official Editorial](https://codeforces.com/blog/entry/81506)
+
+<LanguageSection>
+
+<CPPSection>
+
+## Solution - Prefix Sums + Math
+
+```cpp
+#include <bits/stdc++.h>
```
Comment (maggie-j-liu) [inline solutions/CF Good Subarrays.mdx:16]: use tabs

File: solutions/CF Good Subarrays.mdx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        dp[i] += dp[i-1];
+    }
+    map<int, ll> sumDist;
+    for(int i=0; i<=n; i++){
+        sumDist[dp[i] - i]++;
+    }
+    ll goodArrays = 0;
+    for(pi p: sumDist){
+        ll f = p.second;
+        goodArrays += ((f) * (f-1)) / 2;
```
Comment (maggie-j-liu) [inline solutions/CF Good Subarrays.mdx:39]:
```suggestion
        goodArrays += f * (f-1) / 2;
```

File: solutions/CF Good Subarrays.mdx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+## Solution - Prefix Sums + Math
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using pi = pair<ll, ll>;
+
+void solve(){
```
Comment (maggie-j-liu) [inline solutions/CF Good Subarrays.mdx:22]: spacing is inconsistent; either space all operators on both sides or don't space any

Comment (maggie-j-liu) [reply solutions/CF Good Subarrays.mdx:20]: like you have `sumDist[dp[i] - i]++;` and also `dp[i-1]` so just make the spacing the same (i didn't mean the newlines)

File: solutions/LC matrix-block-sum.mdx [hunk 5]
```diff
@@ -0,0 +1,55 @@
+---
```
Comment (maggie-j-liu) [inline solutions/LC matrix-block-sum.mdx:1]: modify the problems json file to add this solution

File: solutions/LC matrix-block-sum.mdx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+**Time Complexity:** $\mathcal O(NM)$.
+
+<LanguageSection>
+
+<CPPSection>
+
+## Implementation
+
+```cpp
+class Solution {
```
Comment (maggie-j-liu) [inline solutions/LC matrix-block-sum.mdx:17]: use tabs

File: solutions/LC matrix-block-sum.mdx [hunk 7]
```diff
@@ -0,0 +1,55 @@
+---
+id: lc-MatrixBlockSum
+source: LeetCode
+title: Matrix Block Sum
+author: Jesse Choe
+---
+
+**Time Complexity:** $\mathcal O(NM)$.
```
Comment (maggie-j-liu) [inline solutions/LC matrix-block-sum.mdx:8]:
```suggestion
**Time Complexity:** $\mathcal O(NM)$
```

File: solutions/LC matrix-block-sum.mdx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+<CPPSection>
+
+## Implementation
+
+```cpp
+class Solution {
+public:
+    vector<vector<int>> matrixBlockSum(vector<vector<int>>& mat, int K) {
+        int n = mat.size(), m = mat[0].size();
+        int arr[n][m];
```
Comment (maggie-j-liu) [inline solutions/LC matrix-block-sum.mdx:21]: variable length arrays are not c++ standard

File: solutions/LC matrix-block-sum.mdx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        }
+        int dp[n+1][m+1];
+        for(int i=0; i<=n; i++){
+            for(int j=0; j<=m; j++){
+                dp[i][j] = 0;
+            }
+        }
+        for(int i=1; i<=n; i++){
+            for(int j=1; j<=m; j++){
+                dp[i][j] = dp[i-1][j] + dp[i][j-1] - dp[i-1][j-1] + arr[i-1][j-1];
```
Comment (maggie-j-liu) [inline solutions/LC matrix-block-sum.mdx:35]: spacing is inconsistent

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/LC matrix-block-sum.mdx (hunk 9 above)
Comment (jessechoe10) [reply solutions/LC matrix-block-sum.mdx:35]: lol. i think u're gonna have to give suggestion. i still don't follow :(

Comment (maggie-j-liu) [reply solutions/LC matrix-block-sum.mdx:35]: `dp[i][j] = dp[i-1][j] + dp[i][j-1] - dp[i-1][j-1] + arr[i-1][j-1];`
There are spaces on both sides of `+` in `dp[i-1][j] + dp[i][j-1]` but there are no spaces on both sides of `-` in `dp[i-1]`

File: solutions/CF Good Subarrays.mdx (hunk 2 above)
Comment (maggie-j-liu) [reply solutions/CF Good Subarrays.mdx:16]: use tabs to indent, not spaces

File: solutions/CF Good Subarrays.mdx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+## Solution - Prefix Sums + Math
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+using pi = pair<ll, ll>;
+void solve(){
+    int n; cin >> n;
+    vector<int> dp(n+1, 0);
```
Comment (andrewwangva) [inline solutions/CF Good Subarrays.mdx:22]: would be better to replace dp with psum, given it's not really dp.

Comment (jessechoe10) [reply solutions/CF Good Subarrays.mdx:22]: Well most of these editorials use `dp`, so I decided to make all of my psum arrays named `dp` lol.

File: solutions/LC matrix-block-sum.mdx (hunk 8 above)
Comment (jessechoe10) [reply solutions/LC matrix-block-sum.mdx:21]: uhhhhhhh i just made it this way b/c i'm not comfortable using vector<vector<int>> lol.

Comment (maggie-j-liu) [reply solutions/LC matrix-block-sum.mdx]: make the sizes of the array the max possible from the bounds of the problem

File: solutions/CF Good Subarrays.mdx (hunk 10 above)
Comment (maggie-j-liu) [reply solutions/CF Good Subarrays.mdx]: i think it would probably be more clear to name them `psum`...

File: solutions/LC matrix-block-sum.mdx [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+                }
+                int dp[n+1][m+1];
+                for(int i=0; i<=n; i++){
+                    for(int j=0; j<=m; j++){
+                        dp[i][j] = 0;
+                    }
+                }
+                for(int i=1; i<=n; i++){
+                    for(int j=1; j<=m; j++){
+                        dp[i][j] =  dp[i-1][j]  +  dp[i][j-1]  -  dp[i-1][j-1]  +  arr[i-1][j-1];
```
Comment (maggie-j-liu) [inline solutions/LC matrix-block-sum.mdx:35]: the spacing is still inconsistent (and now there are two spaces???) let me know if it's still unclear what you need to fix.

File: solutions/LC matrix-block-sum.mdx [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+                    }
+                }
+                vector<vector<int>> answer;
+                for(int i=1; i<=n; i++){
+                    vector<int> cur;
+                    for(int j=1; j<=m; j++){
+                        int a = max(1, i - K), b = max(1, j - K), c = min(n, i + K), d = min(m, j + K);
+                        int blockSum =  dp[c][d]  -  dp[c][b-1]  -  dp[a-1][d]  +  dp[a-1][b-1]
+                        cur.push_back( blockSum );
```
Comment (maggie-j-liu) [inline solutions/LC matrix-block-sum.mdx:44]:
```suggestion
                        cur.push_back(blockSum);
```

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/LC matrix-block-sum.mdx (hunk 11 above)
Comment (jessechoe10) [reply solutions/LC matrix-block-sum.mdx:35]: Still unclear lol. Can you suggest change??

File: solutions/CF Good Subarrays.mdx (hunk 10 above)
Comment (jessechoe10) [reply solutions/CF Good Subarrays.mdx]: idt it matters lol.

File: solutions/LC matrix-block-sum.mdx (hunk 11 above)
Comment (maggie-j-liu) [reply solutions/LC matrix-block-sum.mdx:35]:
```suggestion
                        dp[i][j] = dp[i - 1][j] + dp[i][j - 1] - dp[i - 1][j - 1] + arr[i - 1][j - 1];
```
do this for everything else, make sure each operator (`+=-/`) is spaced with exactly one space on each side.

File: solutions/LC matrix-block-sum.mdx [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+<CPPSection>
+
+## Implementation
+
+```cpp
+class Solution {
+    public:
+            vector<vector<int>> matrixBlockSum(vector<vector<int>>& mat, int K) {
+                int n = mat.size(), m = mat[0].size();
+                int arr[n][m];
```
Comment (frodakcin) [inline solutions/LC matrix-block-sum.mdx:21]: I remember something about this not being cpp standard -- use 2D vector or static array size.

Comment (frodakcin) [reply solutions/LC matrix-block-sum.mdx:21]: Likewise for dp array below

File: solutions/LC matrix-block-sum.mdx (hunk 11 above)
Comment (frodakcin) [reply solutions/LC matrix-block-sum.mdx:35]:
```suggestion
						dp[i][j] = dp[i-1][j] + dp[i][j-1] - dp[i-1][j-1] + arr[i-1][j-1];
```

File: solutions/LC matrix-block-sum.mdx [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+                    for(int j=1; j<=m; j++){
+                        dp[i][j] =  dp[i-1][j]  +  dp[i][j-1]  -  dp[i-1][j-1]  +  arr[i-1][j-1];
+                    }
+                }
+                vector<vector<int>> answer;
+                for(int i=1; i<=n; i++){
+                    vector<int> cur;
+                    for(int j=1; j<=m; j++){
+                        int a = max(1, i - K), b = max(1, j - K), c = min(n, i + K), d = min(m, j + K);
+                        int blockSum =  dp[c][d]  -  dp[c][b-1]  -  dp[a-1][d]  +  dp[a-1][b-1]
```
Comment (frodakcin) [inline solutions/LC matrix-block-sum.mdx:43]: This is missing a semicolon?!
(wait is this tested?)

Oh, also fix the spacing.

**Review by frodakcin — CHANGES_REQUESTED** [review]

File: solutions/LC matrix-block-sum.mdx (hunk 11 above)
Comment (frodakcin) [reply solutions/LC matrix-block-sum.mdx:35]: Didn't see you alr suggested.

File: solutions/LC matrix-block-sum.mdx (hunk 14 above)
Comment (jessechoe10) [reply solutions/LC matrix-block-sum.mdx:43]: Oh shoot mb

File: content/3_Silver/Prefix_Sums.problems.json [hunk 15]
```diff
@@ -133,8 +133,7 @@
       "isStarred": false,
       "tags": ["Prefix Sums", "Math"],
       "solutionMetadata": {
-        "kind": "autogen-label-from-site",
-        "site": "CF"
+        "kind": "internal"
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums.problems.json:136]: use CF for kind and site should be the problem URL on CF

File: content/3_Silver/Prefix_Sums_2.problems.json [hunk 16]
```diff
@@ -65,7 +65,7 @@
       "isStarred": false,
       "tags": ["2D Prefix Sums"],
       "solutionMetadata": {
-        "kind": "none"
+        "kind": "internal"
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.problems.json:68]: same here

File: solutions/CF Good Subarrays.mdx [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+title: Good Subarrays
+author: Jesse Choe
+---
+[Official Editorial](https://codeforces.com/blog/entry/81506)
+
+<LanguageSection>
+
+<CPPSection>
+
+## Solution - Prefix Sums + Math
```
Comment (HarryWangATX) [inline solutions/CF Good Subarrays.mdx:13]: bring outside

File: solutions/LC matrix-block-sum.mdx [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+author: Jesse Choe
+---
+
+**Time Complexity:** $\mathcal O(NM)$
+
+<LanguageSection>
+
+<CPPSection>
+
+## Implementation
```
Comment (HarryWangATX) [inline solutions/LC matrix-block-sum.mdx:14]: same here

File: solutions/LC matrix-block-sum.mdx (hunk 13 above)
Comment (HarryWangATX) [reply solutions/LC matrix-block-sum.mdx:21]: yes this is not cpp standard use vector

File: solutions/LC matrix-block-sum.mdx [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+    public:
+            vector<vector<int>> matrixBlockSum(vector<vector<int>>& mat, int K) {
+                int n = mat.size(), m = mat[0].size();
+                int arr[n][m];
+                for(int i=0; i<n; i++){
+                    for(int j=0; j<m; j++){
+                        arr[i][j] = mat[i][j];
+                    }
+                }
+                int dp[n+1][m+1];
```
Comment (HarryWangATX) [inline solutions/LC matrix-block-sum.mdx:27]: same here

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.problems.json (hunk 16 above)
Comment (HarryWangATX) [reply content/3_Silver/Prefix_Sums_2.problems.json:68]: mb i misread

File: content/3_Silver/Prefix_Sums.problems.json (hunk 15 above)
Comment (HarryWangATX) [reply content/3_Silver/Prefix_Sums.problems.json:136]: nvm mb i misread

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]
Use tabs instead of spaces to indent. The spacing around operators is still inconsistent; if you  prefer to use no spaces, make sure it's the same for all operators `+-*/=`.

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]
> Use tabs instead of spaces to indent.

**Comment by jessechoe10** [comment]
> > Use tabs instead of spaces to indent.

I did...

**Comment by HarryWangATX** [comment]
> > > Use tabs instead of spaces to indent.
> 
> I did...

not from what I'm seeing

**Comment by jessechoe10** [comment]
> > > > Use tabs instead of spaces to indent.
> > 
> > 
> > I did...
> 
> not from what I'm seeing

give example of where i didn't...

**Comment by HarryWangATX** [comment]
> > > > > Use tabs instead of spaces to indent.
> > > 
> > > 
> > > I did...
> > 
> > 
> > not from what I'm seeing
> 
> give example of where i didn't...

literally the whole file...

**Comment by jessechoe10** [comment]
> > > > > > Use tabs instead of spaces to indent.
> > > > 
> > > > 
> > > > I did...
> > > 
> > > 
> > > not from what I'm seeing
> > 
> > 
> > give example of where i didn't...
> 
> literally the whole file...

commit suggestion, then. I don't see it 😕

**Comment by HarryWangATX** [comment]
@maggie-j-liu i think u added a lil too much lol

**Comment by jessechoe10** [comment]
@HarryWangATX and @maggie-j-liu does everything else look good rn. if so, can u approve 😛

**Comment by HarryWangATX** [comment]
not as of now xD, maggie added too much tabs after i commited

**Comment by jessechoe10** [comment]
> not as of now xD, maggie added too much tabs after i committed

oof 🤦‍♂️

**Comment by HarryWangATX** [comment]
Wait, i might have misread diff file. i think maggie just redid the commit or smth

**Comment by HarryWangATX** [comment]
oh u have 2 sols myb

**Review by HarryWangATX — APPROVED** [review]

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/LC matrix-block-sum.mdx [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+			for(int i=1; i<=n; i++){
+				for(int j=1; j<=m; j++){
+					dp[i][j]=dp[i-1][j]+dp[i][j-1]-dp[i-1][j-1]+arr[i-1][j-1];
+				}
+			}
+			vector<vector<int>> answer;
+			for(int i=1; i<=n; i++){
+				vector<int> cur;
+				for(int j=1; j<=m; j++){
+					int a=max(1, i-K), b=max(1, j-K), c=min(n, i+K), d=min(m, j+K);
```
Comment (maggie-j-liu) [inline solutions/LC matrix-block-sum.mdx:42]:
```suggestion
					int a=max(1, i-K), b=max(1, j-K);
					int c=min(n, i+K), d=min(m, j+K);
```
this line is too long

**Comment by jessechoe10** [comment]
@frodakcin pls review.

**Review by frodakcin — APPROVED** [review]

**Pull request merged**